### ISSUE TITLE
iOS - Pulling and pushing plural strings dict creates a duplicate 

### ISSUE THREAD

**Issue by teameh** [phrase/phraseapp-client#199]
Pulling and pushing plurals with a strings dict creates a duplicate key with "||localized_key_format" appended. 

I start off the following (plural) copy in Phrase:

<img width="582" alt="image" src="https://user-images.githubusercontent.com/1330668/82313550-cc668080-99c8-11ea-88c1-8a74940d63e3.png">

#### Pull
```
$ phraseapp pull
Downloaded EN to Localizable.stringsdict
```

#### Result: 
```
$ cat Localizable.stringsdict

<?xml version="1.0" encoding="UTF-8"?>
<plist version="1.0">
  <dict>
    <key>my_plural_bicycles_key</key>
    <dict>
      <key>NSStringLocalizedFormatKey</key>
      <string>%#@localized_format_key@</string>
      <key>localized_format_key</key>
      <dict>
        <key>NSStringFormatSpecTypeKey</key>
        <string>NSStringPluralRuleType</string>
        <key>NSStringFormatValueTypeKey</key>
        <string>d</string>
        <key>one</key>
        <string>bicycle</string>
        <key>other</key>
        <string>bicycles</string>
      </dict>
    </dict>
  </dict>
</plist>
```

#### Push: 
```
$ phraseapp push
Uploading Localizable.stringsdict... done!
```

#### Pull again: 
```
$ phraseapp pull
Downloaded EN to Localizable.stringsdict
```

#### Result: 

The key is now duplicated: 

```
$ cat RPX/Base.lproj/Localizable.stringsdict

<?xml version="1.0" encoding="UTF-8"?>
<plist version="1.0">
  <dict>
    <key>my_plural_bicycles_key</key>
    <dict>
      <key>NSStringLocalizedFormatKey</key>
      <string>%#@localized_format_key@</string>
      <key>localized_format_key</key>
      <dict>
        <key>NSStringFormatSpecTypeKey</key>
        <string>NSStringPluralRuleType</string>
        <key>NSStringFormatValueTypeKey</key>
        <string>d</string>
        <key>one</key>
        <string>bicycle</string>
        <key>other</key>
        <string>bicycles</string>
      </dict>
      <key>localized_format_key</key>
      <dict>
        <key>NSStringFormatSpecTypeKey</key>
        <string>NSStringPluralRuleType</string>
        <key>NSStringFormatValueTypeKey</key>
        <string>d</string>
        <key>one</key>
        <string>bicycle</string>
        <key>other</key>
        <string>bicycles</string>
      </dict>
    </dict>
  </dict>
</plist>
```

Also in Phrase:

<img width="431" alt="image" src="https://user-images.githubusercontent.com/1330668/82313306-74c81500-99c8-11ea-9a8a-5b4a71701de2.png">

**Comment by rlisowski** [comment]
@teameh thanks for the report, I added it to our backlog

**Comment by cjurjiu** [comment]
@rlisowski also to jump in here...we're also encontering this issue and it breaks sharing strings between iOS & Android. `some_key||localized_key_format` is an invalid key for a string in Android. so we need to manually update the keys to remove `||localized_key_format` each time after pulling the strings.

would be highly appreciated for this to be addressed

**Comment by teameh** [comment]
Yes, that is how we've found out about this too. I think this was introduced in the last version or the previous version of the cli, we had no problems with this before updating

**Comment by ffittschen** [comment]
We have the same issue. I get that Phrase adds the `||formatString` suffix when uploading `.stringsdict` files, because they need some way to "store" the format string so that pulling the translations will result in the same structure as the one that was pushed. But this should be a property in the data model of the key and not _part of the key_. This way it would also be possible to edit the `NSLocalizedFormatString` in the Phrase UI.

Another point why this should be an editable property in the data model is that the format string can contain multiple variables, as in `%#@files@ (%#@bytes@, %#@minutes@)`, which is an example that is explicitly mentioned in Apple's documentation on the [Stringsdict File Format](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/StringsdictFileFormat/StringsdictFileFormat.html).

**Comment by rlisowski** [comment]
I checked and it's API/upload problem, not a  CLI client problem. I can fix `||localized_format_key` problem for you but if you put any other format string there it will be used as usual. Also if you have `bla_bla||localized_format_key` key names you need to edit them to be `bla_bla` keys. 

@ffittschen  I will request to get rid of `||formatString` from key names altogether and move it to editable metadata but it's a bigger topic.

**Comment by rlisowski** [comment]
@teameh let me know if it solves your issue

**Comment by teameh** [comment]
Yes, when pushing now I do not get the duplicate any more. 

Regarding @ffittschen's comment. Is it possible using phrase to add multiple placeholders to a single Stringsdict entry? 

And regarding the interface online. If I enter `bicycles` here: 

![image](https://user-images.githubusercontent.com/1330668/83236824-721daa80-a194-11ea-8d5b-99c635a66536.png)

I expect the .stringsdict to look like: 

```
<?xml version="1.0" encoding="UTF-8"?>
<plist version="1.0">
  <dict>
    <key>my_plural_bicycles_key</key>
    <dict>
      <key>NSStringLocalizedFormatKey</key>
      <string>%#@bicycles@</string>              // <-- 'bicycles'
      <key>bicycles</key>                        // <-- 'bicycles'
      <dict>
        <key>NSStringFormatSpecTypeKey</key>
        <string>NSStringPluralRuleType</string>
        <key>NSStringFormatValueTypeKey</key>
        <string>d</string>
        <key>one</key>
        <string>bicycle</string>
        <key>other</key>
        <string>bicycles</string>
      </dict>
    </dict>
  </dict>
</plist>
```

At least that is what you expect when reading the Stringsdict docs: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/StringsdictFileFormat/StringsdictFileFormat.html 

Is that correct?

**Comment by ffittschen** [comment]
@teameh Do you mean using multiple variables in a plural entry, e.g. setting the `NSStringLocalizedFormatKey` to `%#@files@ (%#@bytes@, %#@minutes@)` as in the Stringsdict File Format you linked?

I would definitely be interested in that as well! Only allowing one variable (and only the variable) is quite limiting. The `NSStringLocalizedFormatKey` should be able to be translated as well and it should also allow other placeholders, because this is the only way to use string placeholders in addition to the numeric placeholders that are used to decide upon the plural rules. One example for this would be 

> NSStringLocalizedFormatKey = "Your %@ list contains %#@elements@.";

The first placeholder is non-numeric, so it can only be used in the `NSStringLocalizedFormatKey` and not as a variable. Some other examples can be found here: https://github.com/SwiftGen/SwiftGen/pull/634

**Comment by teameh** [comment]
No. Although that is also interesting, it's not what I mean. 

I mean that when I enter `bicycles` here: 

![image](https://user-images.githubusercontent.com/1330668/83236824-721daa80-a194-11ea-8d5b-99c635a66536.png)

I expect the .stringsdict to include that variable name `bicycles`:

```
<?xml version="1.0" encoding="UTF-8"?>
<plist version="1.0">
  <dict>
    <key>my_plural_bicycles_key</key>
    <dict>
      <key>NSStringLocalizedFormatKey</key>
      <string>%#@bicycles@</string>              // <-- 'bicycles'
      <key>bicycles</key>                        // <-- 'bicycles'
      <dict>
        <key>NSStringFormatSpecTypeKey</key>
        <string>NSStringPluralRuleType</string>
        <key>NSStringFormatValueTypeKey</key>
        <string>d</string>
        <key>one</key>
        <string>bicycle</string>
        <key>other</key>
        <string>bicycles</string>
      </dict>
    </dict>
  </dict>
</plist>
```

Instead I get:

```
<?xml version="1.0" encoding="UTF-8"?>
<plist version="1.0">
  <dict>
    <key>my_plural_bicycles_key</key>
    <dict>
      <key>NSStringLocalizedFormatKey</key>
      <string>%#@localized_format_key@</string>    // <-- 'localized_format_key'
      <key>localized_format_key</key>              // <-- 'localized_format_key'
      <dict>
        <key>NSStringFormatSpecTypeKey</key>
        <string>NSStringPluralRuleType</string>
        <key>NSStringFormatValueTypeKey</key>
        <string>d</string>
        <key>one</key>
        <string>bicycle</string>
        <key>other</key>
        <string>bicycles</string>
      </dict>
    </dict>
  </dict>
</plist>
```

**Comment by teameh** [comment]
@rlisowski the initial issue is back on the api, we have the same problem as before.
